runtime/src/consensus/registry.rs

```diff
@@ -4,27 +4,323 @@
 //!
 //! This **MUST** be kept in sync with go/registry/api.
 //!
+use serde::{Deserialize, Serialize};
+use serde_repr::*;
+
 use crate::{
-    common::crypto::{hash, signature::SignatureBundle},
+    common::{
+        crypto::{
+            hash::Hash,
+            signature::{PublicKey, SignatureBundle},
+        },
+        namespace::Namespace,
+        quantity,
+        version::Version,
+    },
+    consensus::staking,
     storage::mkvs::WriteLog,
 };
-use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+/// RuntimeKind represents the runtime functionality.
```

Comment:
No need to include the type/field name in rust doc :-)

Reply:
Oops, fixed :)